core/src/main/java/org/web3j/tx/gas/DynamicEIP1559GasProvider.java

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright 2025 Web3 Labs Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package org.web3j.tx.gas;
+
+import java.io.IOException;
+import java.math.BigDecimal;
+import java.math.BigInteger;
+
+import org.web3j.protocol.Web3j;
+import org.web3j.protocol.core.DefaultBlockParameterName;
+import org.web3j.protocol.core.methods.request.Transaction;
+import org.web3j.protocol.core.methods.response.EthEstimateGas;
+import org.web3j.protocol.core.methods.response.EthGasPrice;
+import org.web3j.protocol.core.methods.response.EthMaxPriorityFeePerGas;
+
+public class DynamicEIP1559GasProvider implements ContractEIP1559GasProvider, PriorityGasProvider {
+    private Web3j web3j;
+    private long chainId;
+    private final Priority priority;
+    private final BigDecimal customMultiplier;
+    public static final BigInteger GAS_LIMIT = BigInteger.valueOf(9_000_000);
+
+    public DynamicEIP1559GasProvider(Web3j web3j, long chainId) {
+        this(web3j, chainId, Priority.NORMAL);
+    }
+
+    public DynamicEIP1559GasProvider(Web3j web3j, long chainId, Priority priority) {
+        this(web3j, chainId, priority, BigDecimal.ONE);
+    }
+
+    public DynamicEIP1559GasProvider(
+            Web3j web3j, long chainId, Priority priority, BigDecimal customMultiplier) {
+        this.web3j = web3j;
+        this.chainId = chainId;
+        this.priority = priority;
+        this.customMultiplier = customMultiplier;
+    }
+
+    @Override
+    public boolean isEIP1559Enabled() {
+        return true;
```

Comment:
as this comes from ContractEIP1559GasProvide when using this gas provider this check is redundant because it returns true by default, it doesn't check if eip1559 is enabled on the network. This method should be removed.

Reply:
ContractEIP1559GasProvider is an interface extending this method isEIP1559Enabled()
This is being implemented by StaticEIP1559GasProvider too and hence in EIP1559DynamicGasProvider also i have implemented this to return true.

Do you mean shall I remove this method from both the classes and in interface default it to true?

Reply:
Makes sense, removed